pkg/apis/batch/validation/validation_test.go

```diff
@@ -824,6 +824,38 @@ func TestValidateJob(t *testing.T) {
 				Template: validPodTemplateSpecForGenerated,
 			},
 		},
+		"spec.template.labels: Invalid value": {
```

Comment:
I'd love to see two different test cases - one with just a long metadata.name (proving it fails when using a generated selector) and one with a manual selector and labels (this).

My suspicion is that this fails because the template fails validation, but I am more worried about it passing template validation, then getting the default selector labels set, then being stored invalid.

Also, this test is not super useful without some checking of WHY the error happened.  It should do a count check (e.g. "I got 1 error, but expected 2") and it should do a substring check ("the error should say 'Invalid: metadata.name' somewhere").  I bet there are a handful of "expected failures" that are for the wrong reasons (there certainly were in other subsystems I have tested this way).

Also, this test is BEGGING for boilerplate reduction, like  https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/storage/storage_test.go#L1459-L1608

Reply:
Your hunch was correct.  I think this is a bug.  

Job-Name was used in our label and we validate it to make sure its a valid label.  That has a limit of 63 characters but we were validating the name with a different requirement (256 characters).  

So it was possible to create a job with job name of 63 characters and you can only get validation fail if you used 256 characters.  

I pushed a fix for this.